### DISCUSSION TITLE
Mojo Logo & Name Usage?

### DISCUSSION THREAD

**Discussion by cirqol** [modular/modular#389] [General]
Will the community be able to use the Mojo name and logo for the purposes of education? Just figured now is the best time clarify considering what's currently going on in the Rust community. 

I'm already working on and planning a Mojo specific YT channel. Just curious what the rules are in regards to the logo and name.

**Comment by lattner** [comment]
Hi @cirqol thanks for raising this, we definitely want the community to be able to use logo and name.  We should get a proper web page up that describes this, @shashankprasanna / @scottamain / @iamtimdavis :)

**Reply by cirqol** [reply]
Thanks for clarifying.

**Comment by goyalyashpal** [comment]
You and others may use this thread to post references (links etc) to the "Logo & Trademark" aspect of other projects.
_oh lol, wish github had "mark as community wiki" option in discussions - just like in stackoverflow_

For example: here's one for:

Foundation | Page Name | Page Link
-- | -- | -- 
Blender | Logo & Trademark | https://www.blender.org/about/logo/
Rust-Lang Foundation | Logo Policy and Media Guide | https://foundation.rust-lang.org/policies/logo-policy-and-media-guide/

**Comment by lattner** [comment]
We're behind on this, but it is mostly getting the details sorted out.  My current understanding:  we want people to be free to use the word Mojo and Mojo🔥, and using the Mojo🔥 logo is fine.  The things we need to protect are: 1) don't represent that you are speaking on behalf of modular, and 2) don't use the "Modular M" with the notch taken out without permission. 

It is fine to use Mojo or Mojo🔥 with a normal M.

**Reply by s3drik** [reply]
Hello. This great using logo but i think Mojo🔥 isn't look like logo. It is a bit boring. All programming languages have their logo and some of them have their symbol for example rust have logo with R and 🦀. there is something similar to the word "go".  I mean, mojo needs something like that.  for example, mojo is a superset of python and uses the fire emoji.  I asked AI to create something like this and I got these pictures (of course this is just an example. I have no design skills so I used ai and sorry for my english)
<img src="https://github.com/modularml/mojo/assets/97380947/247d7c5f-ab29-4d8a-a503-45b5b15e526a" width="275px"/>
<img src="https://github.com/modularml/mojo/assets/97380947/f3ea6da2-e3bc-47d7-84cd-203ea6262d7a" width="275px"/>
<img src="https://github.com/modularml/mojo/assets/97380947/a592a6af-b5f0-4f87-abf3-304336e2d4c5" width="275px"/>
<img src="https://github.com/modularml/mojo/assets/97380947/c0dc7ca9-f999-4f9c-b6ba-ee492d2cb554" width="275px"/>
<img src="https://github.com/modularml/mojo/assets/97380947/d3094341-48f0-4200-88da-b17a816d50cc" width="275px"/>

**Reply by goyalyashpal** [reply]
> when you send screenshots, could you please make them properly sized? 
> Use `<img src="INSERT_LINK_HERE" width="275px" />`
> _- https://github.com/TeamNewPipe/NewPipe/issues/6841#issuecomment-892756057_

**Reply by goyalyashpal** [reply]
> <img src="https://user-images.githubusercontent.com/97380947/248861347-f3ea6da2-e3bc-47d7-84cd-203ea6262d7a.jpg" width="125px"/>

wow, this looks nice - Plain style, plenty of whitespace

If there is will, then i will love to contribute to make it suit Mojo (instead of Jojo lol)

**Comment by iamtimdavis** [comment]
That's right 👆🏼 

We've also seen a lot of the troubles of other communities, and want to ensure that the Mojo🔥 community has a clear understanding of our trademark rights, and the relevant community usage from the beginning. 

The spirit of what we want to achieve is essentially to have a "Community Logo" and a "Official Logo" that enables a flexible use for the community, but also provides us with an ability to have "Official Use" when needed. There will be subtle differences (i.e. the Notch in the M, the style of the Fire icon etc) but enabling our incredible community to use the logo is definitely our goal 👍🏼

**Comment by nav9** [comment]
Jeremy Howard had referred to Mojo as a "Python++". I believe that's a far better name than Mojo. Unfortunately [Py++ is taken](https://pypi.org/project/pyplusplus/). But naming Mojo as `Python++` would make it far more relatable (an instant mental connect that it's a superset of Python), popular and it'll improve SEO. As of today a search for "Mojo" leads to various sites named Mojo. I ran into this problem when Google searching for Mojo's documentation.  
Additionally, I hope y'all are actively consulting end users about the various new types of syntax y'all are introducing into Mojo? Python became popular due to its simple syntax. I'm a supporter of Mojo, but the various complicated new syntax y'all have brought into Mojo, has been a turn-off for me, in the same way that I stopped learning `Julia`. Ideally, the syntax should be simple, and the complications should be handled under the hood. I know it's easier said than done, but it helps to be Uncle Scrooge, whenever someone on the team suggests using a new symbol for some syntax or semantics. Every new component becomes a burden for developers to learn.  
Great work though. Especially the idea to create a faster Python, and all the other work which makes deployment easy.

**Comment by ksandvik** [comment]
Just my options:
a) C++ introduced syntax that broke C code, Mojo intends to be pythonic so the same syntax will work. Hence a ++ naming was not even a good idea for C++.  SEO optimizations will happen so using Mojo as a name is not an issue.
b) As Mojo will be 1:1 with Python syntax anyone could code from get go with this syntax. Systems level programming requires additional syntax that is hard to make easy as systems programming itself is not easy. But I'm sure there will be an ease-in way to learn the additional syntax via documentation and samples.

**Comment by nav9** [comment]
@ksandvik : Sure, I respect that y'all would like to keep the name as Mojo. As for syntax, I do hope the new introductions like [inout](https://github.com/modularml/mojo/issues/7), `owned` (with a `^` postfix) etc. are planned well and add value as an improvement, because [familiarity matters](https://nav9.medium.com/when-can-we-use-spoken-english-as-a-programming-language-8aa30f2e009c) in syntax, as much as in user interfaces (saw [this](https://www.reddit.com/r/ProgrammerHumor/comments/14ykcoi/idontwanttolearnrust/) just yesterday). Personally, what I'm looking forward to in Mojo, is the ability [to use it locally](https://github.com/modularml/mojo/discussions/144#discussioncomment-5918668), and to use any GPU ([AMD appears to need help with it](https://github.com/RadeonOpenCompute/ROCm/issues/2216)). Thank you for this wonderful framework, and the thought process behind it.

**Comment by lattner** [comment]
Just to set expectations here, we don't plan to rename Mojo - we quite like the name :-).  I agree with you that Python++ is a useful working model to think about some of Mojo's goals though!

**Reply by martin-braun** [reply]
How will you address the ambiguity between Mojo, the Python superset, and mojo.js, the JS web framework, please? How can "web development using mojo" not cause issues in search engines? You are most likely to compete with the framework when it comes to SEO on "web development" keywords.

I doubt Mojo will be limited to the AI realm, you will conquer the web, frameworks like Django will support Mojo. It will be rough to claim the "mojo web development" terms when it comes to web indexing.

**Reply by goyalyashpal** [reply]
> Mojo vs mojo.js



other things combat with that situation by having an official full name made of the name + categorical prefix...

so, "Mojo lang" and "mojo.js framework"

> Tectonic
> About the name ... In cases where the name might lead to ambiguities, it should be expanded to “Tectonic typesetting.”
>
> \- https://tectonic-typesetting.github.io/en-US/


> Hy
> Hy (or "Hylang" for long) is a ...
>
> \- http://hylang.org/

> ConTeXt
> \- https://wiki.contextgarden.net/Main_Page
>
> ConTeXt typesetting

**Comment by ksandvik** [comment]
There will always be name clashes, can't do much if there's a for example new name and it already exists or will exist. Usually this works out the way that the most relevant, high frequency name, bubbles up first in searches et rest, so I would not worry about this.

**Reply by martin-braun** [reply]
Example in action: https://github.com/TranscryptOrg/Transcrypt/discussions/833#discussioncomment-6865369

This is a Python related project and people aren't sure what I'm talking about (yet). We will see how things will turn out over the next months or years with that name.
